src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -12,11 +13,14 @@ namespace System.Net.Http
     internal class WinHttpResponseStream : Stream
     {
         private volatile bool _disposed = false;
-        private SafeWinHttpHandle _requestHandle = null;
+        private WinHttpRequestState _state = null;
```

Comment:
> Won't the compiler take care of optimizing things like this? 

Yes.

> I can make those change you suggest but sometimes it is valuable when reading code to have all field initializes in one place.

It's up to you.  Personally for me I find it harder to internalize code that sets things to default values, especially when the code overwrites it immediately but in a different place in the code.